code/datums/abilities/vampire/phaseshift_vampire.dm

```diff
@@ -9,7 +9,7 @@
 	pointCost = 0
 	when_stunned = 1
 	not_when_handcuffed = 0
-	restricted_area_check = 1
+	restricted_area_check = 0
```

Comment:
Huh... Why didn't this work?
https://github.com/goonstation/goonstation/blob/17d2fc3f2b4b26385f9338fa89e05576fc792cdb/code/datums/abilities/ability_parent.dm#L940

Reply:
The restricted area check was for the "toggle bat form" ability button. All that ability does is alter your mob's `special_sprint` flags (adding/removing either `SPRINT_BAT` or `SPRINT_BAT_CLOAKED`). This is what caused the behavior reported in #9806 - as a vampire you can't toggle whether sprinting should or should not bat form while in restricted z-space.

That property is checked when you start sprinting: https://github.com/goonstation/goonstation/blob/master/code/mob/living.dm#L1758

Both properties end up calling the same `spell_batpoof` proc, where the new z-level restriction check is placed.